group-coordinator/src/test/java/org/apache/kafka/coordinator/group/OffsetMetadataManagerTest.java

```diff
@@ -3050,6 +3060,53 @@ public void testOffsetDeletionsSensor() {
         verify(context.metrics).record(OFFSET_DELETIONS_SENSOR_NAME, 2);
     }
 
+    @Test
+    public void testOnPartitionsDeleted() {
+        OffsetMetadataManagerTestContext context = new OffsetMetadataManagerTestContext.Builder().build();
+
+        // Commit offsets.
+        context.commitOffset("grp-0", "foo", 1, 100, 1, context.time.milliseconds());
+        context.commitOffset("grp-0", "foo", 2, 200, 1, context.time.milliseconds());
+        context.commitOffset("grp-0", "foo", 3, 300, 1, context.time.milliseconds());
+
+        context.commitOffset("grp-1", "bar", 1, 100, 1, context.time.milliseconds());
+        context.commitOffset("grp-1", "bar", 2, 200, 1, context.time.milliseconds());
+        context.commitOffset("grp-1", "bar", 3, 300, 1, context.time.milliseconds());
+
+        context.commitOffset(100L, "grp-2", "foo", 1, 100, 1, context.time.milliseconds());
+        context.commitOffset(100L, "grp-2", "foo", 2, 200, 1, context.time.milliseconds());
+        context.commitOffset(100L, "grp-2", "foo", 3, 300, 1, context.time.milliseconds());
+
+        // Delete partitions.
+        List<Record> records = context.deletePartitions(Arrays.asList(
+            new TopicPartition("foo", 1),
```

Comment:
any particular reason why this is tested with 3 partitions per topic?

Reply:
Not really. I wanted more than one so I used three ;).

Reply:
So in the case where we delete one partition for the bar topic, that wouldn't necessarily happen (i don't think we support deleting just one partition of a topic)

Reply:
Yeah, the code is actually written to support deleting one partition even if we don't support it.